Sources/FoundationEssentials/Subprocess/Subprocess+API.swift

```diff
@@ -0,0 +1,225 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import SystemPackage
+
+extension Subprocess {
+    public static func run(
+        executing executable: Executable,
+        arguments: Arguments = [],
+        environment: Environment = .inherit,
+        workingDirectory: FilePath? = nil,
+        platformOptions: PlatformOptions = .default,
```

Comment:
Do we really need all of those methods that don't take a configuration? To me it seems simpler and cleaner if we just had methods that take a `configuration` and then overload based on input/output/error

Reply:
It's the other way around. These are intended to be the "main" APIs. I only exposed the `configuration` based one for the use case where someone might want to "store" a configuration somewhere and use it later. Otherwise they should always use theses APIs.